### DISCUSSION TITLE
Suggestion: show a QR code for the URL

### DISCUSSION THREAD

**Discussion by wehkah** [Kozea/Radicale#1453] [Ideas]
Hello,

I did not know where to put it, so I put my suggestion here.

When I set up Radicale2 on my server, I was looking for a way to transfer the strange looking URL, that contains this weird looking UID, from my desktop monitor to my Android phone, and I thought that this might be a good reason to include a QR code next to the URL, that can be scanned by the smartphone's camera.

By the way, while we are talking about this UID as part of the URL: no offence meant, but could some one explain to a mere user why the URL that points to the calendar does have to look like that? I am sure there must be a much more smoother way to access ones calendar...  ;)

**Comment by Unrud** [comment]
It depends on the client you are using. Most clients are able to automatically discover your calendars and address books given the server address, username and password.

> I am sure there must be a much more smoother way to access ones calendar... ;)

If you don't like the address go to the storage folder and rename the directory.

**Comment by wehkah** [comment]
> If you don't like the address go to the storage folder and rename the directory.

If it is that easy to get a more self explaining URL for a calendar or an address book, then the obvious question is: why doesn't the program use the name of a calendar as folder name, so that the resulting URL would look like this:

https://server/user/calendar-name/

What has once been the reason for using a random string as folder name in the first place?

**Comment by pbiering** [comment]
> If it is that easy to get a more self explaining URL for a calendar or an address book, then the obvious question is: why doesn't the program use the name of a calendar as folder name, so that the resulting URL would look like this:
> 
> https://server/user/calendar-name/
> 
> What has once been the reason for using a random string as folder name in the first place?

Default is random ID, but in upcoming version 3.2 new calendars can be configured using a custom name via WebUI.

https://github.com/Kozea/Radicale/discussions/1416

Existing ones have to be renamed by renaming directory on file system level.
